crates/bevy_pbr/src/render/pbr_prepass.wgsl

```diff
@@ -47,9 +49,7 @@ fn prepass_alpha_discard(in: FragmentInput) {
     if ((material.flags & STANDARD_MATERIAL_FLAGS_ALPHA_MODE_MASK) != 0u) && output_color.a < material.alpha_cutoff {
         discard;
     }
-#endif // ALPHA_MASK
-
-#ifdef BLEND_PREMULTIPLIED_ALPHA
+#else // BLEND_PREMULTIPLIED_ALPHA || BLEND_ALPHA
     let alpha_mode = material.flags & STANDARD_MATERIAL_FLAGS_ALPHA_MODE_RESERVED_BITS;
     if (alpha_mode == STANDARD_MATERIAL_FLAGS_ALPHA_MODE_BLEND || alpha_mode == STANDARD_MATERIAL_FLAGS_ALPHA_MODE_ADD) 
```

Comment:
Hmm I _think_ this will needleslly run in `BLEND_MULTIPLY`, maybe we can guard it with a `#ifndef BLEND_MULTIPLY`?

Reply:
i think it's ok as is, since if we have `BLEND_MULTIPLY`, which implies we have `!BLEND_PREMULTIPLIED_ALPHA && !BLEND_ALPHA && !ALPHA_MASK`, then `EMPTY_PREPASS_ALPHA_DISCARD` gets defined and this whole containing block is skipped.

i think this set of defs could use a bit of a rationalization at some point though, it seems complex for what it's actually doing.

Reply:
Yeah. Probably better to do that after 0.10 is out to avoid subtly breaking anything.

I wanna try adding a dither-based alpha mode that renders “transparency” in the opaque pass (would go well with TAA) so maybe that could be a good time to do that